src/rgw/rgw_op.cc

```diff
@@ -1089,6 +1091,33 @@ int RGWGetObj::get_data_cb(bufferlist& bl, off_t bl_ofs, off_t bl_len)
     gc_invalidate_time = start_time;
     gc_invalidate_time += (s->cct->_conf->rgw_gc_obj_min_wait / 2);
   }
+  // compression stuff
+  RGWCompressionInfo cs_info;
+  bool need_decompress;
+  int op_ret = rgw_compression_info_from_attrset(attrs, need_decompress, cs_info);
```

Comment:
Why the attribute is decoded multiple times? I saw a call to ```rgw_compression_info_from_attrset``` from ```RGWGetObj::execute```. It looks that turning ```cs_info``` and ```need_decompress``` into class' members would be enough.